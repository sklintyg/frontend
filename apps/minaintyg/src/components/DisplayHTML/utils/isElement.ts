import type { DOMNode } from 'html-react-parser'
import { Element } from 'html-react-parser'

export const isElement = (node: Element | DOMNode | ChildNode): node is Element => node instanceof Element
