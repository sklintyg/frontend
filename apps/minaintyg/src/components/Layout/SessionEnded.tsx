import { IDSIconChevron, IDSIconExternal, IDSLink } from '@frontend/ids-react-ts'
import { Link } from 'react-router-dom'
import { PageHero } from '../PageHero/PageHero'
import { PageHeroActions } from '../PageHero/PageHeroActions'

export function SessionEnded() {
  return (
    <PageHero heading="Du 채r utloggad" type="success">
      <p className="ids-preamble">
        Du har antingen valt att logga ut eller blivit utloggad p책 grund av inaktivitet. Du kan nu st채nga fliken eller logga in p책 nytt.
      </p>
      <PageHeroActions>
        <IDSLink>
          <IDSIconChevron />
          <Link to={import.meta.env.VITE_LOGIN_URL}>Till inloggning</Link>
        </IDSLink>
        <IDSLink>
          <IDSIconChevron />
          <Link to="https://www.1177.se">Till 1177</Link>
          <IDSIconExternal slot="append-icon" />
        </IDSLink>
      </PageHeroActions>
    </PageHero>
  )
}
