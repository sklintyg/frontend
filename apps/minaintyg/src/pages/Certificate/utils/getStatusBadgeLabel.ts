import type { CertificateStatus} from '../../../schema/certificate.schema';
import { CertificateStatusEnum } from '../../../schema/certificate.schema'

export function getStatusBadgeLabel(status: CertificateStatus) {
  switch (status) {
    case CertificateStatusEnum.enum.NEW:
      return 'Nytt'
    case CertificateStatusEnum.enum.REPLACED:
      return 'Ersatt'
    case CertificateStatusEnum.enum.SENT:
      return 'Skickat'
    case CertificateStatusEnum.enum.NOT_SENT:
      return 'Ej skickat'
    default:
      return undefined
  }
}
