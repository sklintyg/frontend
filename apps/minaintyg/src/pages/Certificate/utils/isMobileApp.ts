export const isMobileApp = window.navigator.userAgent.includes('1177-appen')
