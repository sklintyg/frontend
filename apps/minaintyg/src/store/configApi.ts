import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
import { AppConfig } from '../schema/appConfig.schema'

export const configApi = createApi({
  reducerPath: 'api',
  baseQuery: fetchBaseQuery({
    baseUrl: '/appconfig/api/',
  }),
  tagTypes: ['User'],
  endpoints: (builder) => ({
    getAppConfig: builder.query<AppConfig, void>({
      query: () => 'app',
    }),
    getCertificateDescription: builder.query<CertificateDescription, void>({
      query: () => 'map',
    }),
  }),
})

export const { useGetAppConfigQuery, useGetCertificateDescriptionQuery } = configApi
