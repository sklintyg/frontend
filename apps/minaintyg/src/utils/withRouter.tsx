import { ReactNode } from 'react'
import { Route, RouterProvider, createMemoryRouter, createRoutesFromElements } from 'react-router-dom'

export const withRouter = (element: ReactNode) => (
  <RouterProvider router={createMemoryRouter(createRoutesFromElements(<Route path="/" element={element} />))} />
)
