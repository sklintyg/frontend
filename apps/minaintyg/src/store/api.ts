import { getCookie } from '@frontend/utils'
import { isAnyOf, isPlainObject } from '@reduxjs/toolkit'
import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react'
import { AvailableFunction, Certificate, CertificateMetadata } from '../schema/certificate.schema'
import { CertificateFilterOptions } from '../schema/certificateListFilter.schema'
import { User } from '../schema/user.schema'
import { CertificateFilterState } from './slice/certificateFilter.slice'

export const api = createApi({
  reducerPath: 'api',
  baseQuery: fetchBaseQuery({
    baseUrl: '/api/',
    prepareHeaders: (headers) => {
      if (getCookie('XSRF-TOKEN')) {
        headers.set('X-XSRF-TOKEN', getCookie('XSRF-TOKEN') ?? '')
      }
      return headers
    },
  }),
  tagTypes: ['User', 'Certificate'],
  endpoints: (builder) => ({
    getUser: builder.query<User, void>({
      query: () => 'user',
      providesTags: ['User'],
    }),
    getCertificates: builder.query<{ content: CertificateMetadata[] }, CertificateFilterState>({
      query: (body) => ({
        url: 'certificate',
        method: 'POST',
        body: Object.fromEntries(Object.entries(body).map(([key, value]) => [key, [value]])),
      }),
      providesTags: (result) =>
        result
          ? [...result.content.map(({ id }) => ({ type: 'Certificate' as const, id })), { type: 'Certificate', id: 'LIST' }]
          : [{ type: 'Certificate', id: 'LIST' }],
    }),
    getCertificatesFilter: builder.query<CertificateFilterOptions, void>({
      query: () => 'filters',
    }),
    getCertificate: builder.query<Certificate & { availableFunctions: AvailableFunction[] }, { id: string }>({
      query: ({ id }) => `certificate/${id}`,
      transformResponse: ({ certificate, availableFunctions }: { certificate: Certificate; availableFunctions: AvailableFunction[] }) => ({
        ...certificate,
        availableFunctions,
      }),
      providesTags: (result) => (result ? [{ type: 'Certificate' as const, id: result.metadata.id }] : []),
    }),
    sendCertificate: builder.mutation<void, { id: string }>({
      query: ({ id }) => ({
        url: `certificate/${id}/send`,
        method: 'POST',
      }),
      invalidatesTags: (result, error, { id }) => (error ? [] : [{ type: 'Certificate', id }]),
    }),
  }),
})

export const { useGetCertificatesQuery, useGetCertificateQuery, useGetCertificatesFilterQuery, useSendCertificateMutation } = api

export const isFulfilledEndpoint = isAnyOf(...Object.values(api.endpoints).map((endpoint) => endpoint.matchFulfilled))
export const isRejectedEndpoint = isAnyOf(...Object.values(api.endpoints).map((endpoint) => endpoint.matchRejected))
export const hasResponse = (o: unknown): o is { response: Response } =>
  isPlainObject(o) && 'response' in o && o.response instanceof Response
