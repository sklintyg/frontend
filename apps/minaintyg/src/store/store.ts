import { configureStore } from '@reduxjs/toolkit'
import { api } from './api'
import { errorMiddleware } from './middleware/error.middleware'
import { reducer } from './reducer'
import { idpMiddleware } from './slice/idpMiddleware'

export const store = configureStore({
  reducer,
  middleware: (getDefaultMiddleware) => getDefaultMiddleware().concat([api.middleware, idpMiddleware, errorMiddleware]),
})

export type AppDispatch = typeof store.dispatch
