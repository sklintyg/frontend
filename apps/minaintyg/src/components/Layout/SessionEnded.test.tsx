import { render, screen } from '@testing-library/react'
import { Provider } from 'react-redux'
import { updateServiceAvailability } from '../../store/slice/session.slice'
import { store } from '../../store/store'
import { withRouter } from '../../utils/withRouter'
import { SessionEnded } from './SessionEnded'

it('Should render as expected', () => {
  const { container } = render(<Provider store={store}>{withRouter(<SessionEnded />)}</Provider>)
  expect(screen.getByRole('heading', { level: 1, name: 'Du 채r utloggad' })).toBeInTheDocument()
  expect(container).toMatchSnapshot()
})

it('Should display unavailable service error', () => {
  store.dispatch(updateServiceAvailability({ ...new Error('unavailable'), id: '152336e7-41a5-4ef7-abae-c2663c143591' }))
  const { container } = render(<Provider store={store}>{withRouter(<SessionEnded />)}</Provider>)
  expect(screen.getByRole('heading', { level: 1, name: 'Tj채nsten 채r inte tillg채nglig just nu' })).toBeInTheDocument()
  expect(container).toMatchSnapshot()
})
