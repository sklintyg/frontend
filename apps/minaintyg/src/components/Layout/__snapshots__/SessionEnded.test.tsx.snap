// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should display unavailable service error 1`] = `
<div>
  <div
    class="ids-content mx-auto flex max-w-[620px] flex-col gap-10 pt-12 text-center md:pt-24"
  >
    <ids-icon-attention
      class="mx-auto ids-icon"
    />
    <div>
      <h1
        class="ids-heading-1 text-center"
      >
        Tjänsten är inte tillgänglig just nu
      </h1>
      <p
        class="ids-preamble"
      >
        Välkommen tillbaka vid senare tillfälle.
      </p>
      <div
        class="border-y border-stone-clear py-5"
      >
        Om problemet kvarstår, spara nedan id och kontakta 
        <ids-link>
          <a
            href="https://www.1177.se/e-tjanster-support"
            rel="noreferrer"
            target="_blank"
          >
            support
          </a>
          <ids-icon-external
            class="ids-icon"
            slot="append-icon"
          />
        </ids-link>
        <br />
        <strong>
          152336e7-41a5-4ef7-abae-c2663c143591
        </strong>
      </div>
    </div>
  </div>
</div>
`;

exports[`Should render as expected 1`] = `
<div>
  <div
    class="ids-content mx-auto flex max-w-[620px] flex-col gap-10 pt-12 text-center md:pt-24"
  >
    <ids-icon-success
      class="mx-auto ids-icon"
    />
    <div>
      <h1
        class="ids-heading-1 text-center"
      >
        Du är utloggad
      </h1>
      <p
        class="ids-preamble"
      >
        Du har antingen valt att logga ut eller blivit utloggad på grund av inaktivitet. Du kan nu stänga fliken eller logga in på nytt.
      </p>
      <div
        class="mb-5 text-left md:text-center"
      >
        <div
          class="mb-5 mt-8 w-[43px] border-b border-stone-clear md:hidden"
        />
        <div
          class="mx-auto inline-flex flex-col gap-2.5 md:flex-row"
        >
          <ids-link>
            <ids-icon-chevron
              class="ids-icon"
            />
            <a
              href="/saml2/authenticate/eleg"
            >
              Till inloggning
            </a>
          </ids-link>
          <ids-link>
            <ids-icon-chevron
              class="ids-icon"
            />
            <a
              href="https://www.1177.se"
            >
              Till 1177
            </a>
            <ids-icon-external
              class="ids-icon"
              slot="append-icon"
            />
          </ids-link>
        </div>
      </div>
    </div>
  </div>
</div>
`;
